python-backend/app/api/parties/party/resources/party_resource.py

```diff
@@ -72,77 +75,62 @@ def get(self, party_guid=None):
             self.raise_error(400, 'Error: {}'.format(e))
 
+        if not party:
+            self.raise_error(400, 'Error: Failed to create party')
+
         party.save()
         return party.json()
 
     @api.expect(parser)
     @requires_role_mine_create
     def put(self, party_guid):
-        data = PartyResource.parser.parse_args()
-        party_exists = Party.find_by_party_guid(party_guid)
-        if not party_exists:
+        data = self.parser.parse_args()
+        existing_party = Party.find_by_party_guid(party_guid)
+        if not existing_party:
             return self.create_error_payload(404, 'Party not found'), 404
-        party_type_code = data.get('type')
-        if party_type_code == PARTY_STATUS_CODE['per']:
-            first_name = data.get('first_name', party_exists.first_name)
-            party_name = data.get('party_name', party_exists.party_name)
-            party_name_exists = Party.find_by_name(first_name, party_name)
-            if party_name_exists:
-                self.raise_error(
-                    400, 'Error: Party with the name: {} {} already exists'.format(
-                        first_name, party_name))
-            party_exists.first_name = first_name
-            party_exists.party_name = party_name
-        elif party_type_code == PARTY_STATUS_CODE['org']:
-            party_name = data.get('party_name', party_exists.party_name)
-            party_name_exists = Party.find_by_party_name(party_name)
-            if party_name_exists:
-                self.raise_error(400,
-                                 'Error: Party with the name: {} already exists'.format(party_name))
-            party_exists.party_name = party_name
-        else:
-            self.raise_error(400, 'Error: Party type is not provided.')
-        party_exists.save()
-        return party_exists.json()
+
+        try:
+            existing_party.party_name        = data.get('party_name') or existing_party.party_name
+            existing_party.email             = data.get('email') or existing_party.email
+            existing_party.phone_no          = data.get('phone_no') or existing_party.phone_no
+            existing_party.party_type_code   = data.get('type') or existing_party.party_type_code
+            existing_party.first_name        = data.get('first_name') or existing_party.first_name
+            existing_party.suite_no          = data.get('suite_no') or existing_party.suite_no
+            existing_party.address_line_1    = data.get('address_line_1') or existing_party.address_line_1
+            existing_party.address_line_2    = data.get('address_line_2') or existing_party.address_line_2
+            existing_party.city              = data.get('city') or existing_party.city
+            existing_party.sub_division_code = data.get('sub_division_code') or existing_party.sub_division_code
+            existing_party.post_code         = data.get('post_code') or existing_party.post_code
```

Comment:
Was going to wait until tech demo, but this is coming up more. 

We are unable to differentiate between a None being intentionally set, or if the value was meant to unchanged. The pattern I think we are going with is... if the key is in the dictionary (form data on put), then take it's value (assuming it passes validation), if it is not included in the form data then leave it unchanged. 

However by default, reqparser.parse_arg() makes a key for all arguments defined with reqparser.add_argument() (even if it isn't in the request body), this means we cannot detect if the kvp was actually received. The solution is to add store_missing=False to the add_argument() at the top of the resource class, then the condition here is, if 'key' in data.keys(), copy the value. the store_missing=False tells reqparser, only make a kvp if it was actually provided in the request body. 

Let me know if that makes sense. I have an example on my current branch.